src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1286,14 +1363,32 @@ public Worker build() {
                 workerStateChangeListener = DEFAULT_WORKER_STATE_CHANGE_LISTENER;
             }
 
-            if(leaseCleanupValidator == null) {
+            if (shardSyncer == null) {
```

Comment:
This will override the behavior of the lease clean up validator if it is already set. Would be better to do this in 2 checks

```java
if (leaseCleanupValidator == null) {
    leaseCleanupValidator = DEFAULT_LEASE_CLEANUP_VALIDATOR;
}

if (shardSyncer == null) {
    shardSyncer = new KinesisShardSyncer(leaseCleanupValidator);
}
```

Reply:
Changing it since it's already been exposed to customers in an earlier release.

I had done this initially but changed it to just basing both object's creation on shardSyncer. leaseCleanupValidator is used only from within ShardSyncer and hence it doesn't make sense to set it separately if we are anyway injecting ShardSyncer.